features/send/impl/src/main/java/com/tangem/features/send/impl/presentation/state/SendStateFactory.kt

```diff
@@ -138,29 +137,29 @@ internal class SendStateFactory(
     }
 
     fun getOnRecipientAddressValidState(value: String, isValidAddress: Boolean): SendUiState {
+        val cryptoCurrencyStatus = cryptoCurrencyStatusProvider()
         val state = currentStateProvider()
         val recipientState = state.recipientState ?: return state
 
-        val isValidMemo = validateMemo(
-            memo = recipientState.addressTextField.value,
-            cryptoCurrency = cryptoCurrencyStatusProvider().currency,
-        )
-        val isAddressInWallet = isNotAddressInWallet(
-            address = value,
-            walletAddresses = walletAddressesProvider(),
-        )
+        val isValidMemo = validateWalletMemoUseCase(
+            memo = recipientState.memoTextField?.value.orEmpty(),
+            network = cryptoCurrencyStatus.currency.network,
+        ).getOrElse { false }
+        val isAddressInWallet = cryptoCurrencyStatus.value.networkAddress?.availableAddresses
+            ?.none { it.value != value } ?: true
```

Comment:
Кажется так получше:
```suggestion
            ?.any { it.value == value } ?: true
```